specs/work_queue_spec.rb

```diff
@@ -1,9 +1,38 @@
 require_relative '../task_queue'
 
+class MockRecreatableTask
+  include TaskQueue::RecreatableTask
+
+  def run!(**params)
+    puts params.to_s
+  end
+
+  def params_to_hash
+    { one_param: "Hello", other_param: "World" }
+  end
+end
+
+class MockOrderedRecreatableTask
+  include TaskQueue::RecreatableTask
+
+  attr_accessor :number
+  def initialize(number: nil)
+    self.number = number
+  end
+
+  def run!(**params)
+    puts params.to_s
+  end
+
+  def params_to_hash
+    { one_param: "Hello", other_param: "World", number: number }
+  end
+end
+
 module TaskQueue
   describe TaskQueue do
     def wait_for_task_to_complete(task: nil)
-      sleep 0.0001 until task.completed
```

Comment:
Did you find issues with this short sleep time?

Reply:
It seems that tests failed randomly due to the small sleep time, I made lot of tests and with the new sleep time does not fail never.

Reply:
Awesome 🎉